data_selector.py

```diff
@@ -122,40 +123,56 @@ def __load_feature(self, feature):
 
                 title = cfg["name"]
 
+                if cfg.get_filter_value():
+                    filter_expr += " and {}='{}'".format(cfg["feature_filter_column"],
+                                                         cfg.get_filter_value())
+
+                    title = cfg.get_filter_value()
+                else:
+                    title = cfg["name"]
+
+                f = None
+                # test if the layer actually contains data
+                for f in data_l.getFeatures(req):
+                    break
+                if f is None:
+                    return
+                uom = cfg.get_uom()
                 if cfg["type"] == "instantaneous":
-                    if cfg.get_filter_value():
-                        filter_expr += " and {}='{}'".format(cfg["feature_filter_column"],
-                                                             cfg.get_filter_value())
-
-                        title = cfg.get_filter_value()
-                    else:
-                        title = cfg["name"]
-
-                    f = None
-                    for f in data_l.getFeatures(req):
-                        pass
-                    if f is None:
-                        return
-                    uom = cfg.get_uom()
-                    data = LayerData(data_l, cfg["event_column"], cfg["value_column"],
-                                     filter_expression=filter_expr, uom=uom)
-                    uom = data.uom()
-
-                if cfg["type"] == "continuous":
-                    uom = cfg["uom"]
-                    fids = [f.id() for f in data_l.getFeatures(req)]
-                    data = FeatureData(data_l, cfg["values_column"], feature_ids=fids,
-                                       x_start_fieldname=cfg["start_measure_column"],
-                                       x_delta_fieldname=cfg["interval_column"])
+                    data = LayerData(
+                        data_l,
+                        cfg["event_column"],
+                        cfg["value_column"],
+                        filter_expression=filter_expr,
+                        uom=uom
+                    )
+                elif cfg["type"] == "cumulative":
```

Comment:
What about the "continuous" type? Is it definitely obsolete? (We might think about diagraphic data, which are still continuous in our current use case...)

Reply:
Thanks ! That should be fixed now